conda-store-server/conda_store_server/orm.py

```diff
@@ -41,6 +42,22 @@
 ARN_ALLOWED_REGEX = re.compile(schema.ARN_ALLOWED)
 
 
+class Worker(Base):
+    """For communicating with the worker process"""
+
+    __tablename__ = "worker"
+
+    id = Column(Integer, primary_key=True)
+
+    # For checking whether the worker is initialized
+    initialized = Column(Boolean, default=False)
+
+    __table_args__ = (
+        # Ensures no duplicates can be added with this combination of fields.
+        UniqueConstraint("initialized", name="_uc_worker"),
```

Comment:
This unique constraint means that if you ever initialize more than one worker, then you get sql errors. You can try this by starting two workers and you will see:
'''
[2024-01-05 09:16:14,152: ERROR/ForkPoolWorker-3] Task task_initialize_worker[62967c9a-aca7-4f00-b056-15ec3a606507] raised unexpected: IntegrityError('(sqlite3.IntegrityError) UNIQUE constraint failed: worker.initialized')
Traceback (most recent call last):
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlite3.IntegrityError: UNIQUE constraint failed: worker.initialized

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/celery/app/trace.py", line 477, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/celery/app/trace.py", line 760, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/chuck/devel/conda-store/conda-store-server/conda_store_server/worker/tasks.py", line 61, in task_initialize_worker
    db.commit()
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 1451, in commit
    self._transaction.commit(_to_root=self.future)
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 829, in commit
    self._prepare_impl()
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 808, in _prepare_impl
    self.session.flush()
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 3444, in flush
    self._flush(objects)
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 3583, in _flush
    with util.safe_reraise():
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 3544, in _flush
    flush_context.execute()
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/unitofwork.py", line 456, in execute
    rec.execute(self)
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/unitofwork.py", line 630, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
    _emit_insert_statements(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/orm/persistence.py", line 1238, in _emit_insert_statements
    result = connection._execute_20(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2124, in _handle_dbapi_exception
    util.raise_(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/home/chuck/.micromamba/envs/conda-store-server/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: worker.initialized
[SQL: INSERT INTO worker (initialized) VALUES (?)]
[parameters: (1,)]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
'''

I suggest a better approach would be to remove the unique constraint and simply query the worker table for initialized = true. If there is at least one row that is returned then would can proceed.